mscore/inspector/inspectorImage.cpp

```diff
@@ -56,10 +62,42 @@ InspectorImage::InspectorImage(QWidget* parent)
 
 void InspectorImage::valueChanged(int idx)
       {
-      InspectorBase::valueChanged(idx);
+      InspectorElementBase::valueChanged(idx);
       setElement();     // DEBUG
       }
 
+//---------------------------------------------------------
+//   lockAspectRatioClicked
+//---------------------------------------------------------
+
+void InspectorImage::lockAspectRatioClicked(bool clicked)
```

Comment:
Should be:

`void InspectorImage::lockAspectRatioClicked(bool checked)`

Also, I suggest that you throw away the aspect ratio (i.e., reset it to a known ”invalid” value such as `0.0`) when `checked` is `false`.

Reply:
It should be `clicked` because that's what Qt uses.

And is there any real-world usage of throwing away the aspect ratio when it is `false`? If not, then I don't the need of making the code more complex.

Reply:
> It should be `clicked` because that's what Qt uses.

No, it should be `checked` because that's [what Qt uses](https://doc.qt.io/qt-5/qabstractbutton.html#clicked) and also because it's what makes sense. Think about it: When this function is called, it's always because the checkbox has been clicked (or otherwise manipulated by the user), but the checkbox is not always checked. That's what the parameter is there to tell you.

> And is there any real-world usage of throwing away the aspect ratio when it is `false`?

It's an additional sanity check, and it will also be useful when saving to the score file (i.e., a value of `0` means don't save it).

Reply:
I'm convinced of the first matter.

For the second matter, I think "additional sanity check" should only take place if it can prevent latent failures or crashes. And as I said, this fix has nothing to do with saving to score file. If a variable is intended to be saved to score, it should be defined in `libmscore` rather than `mscore`.